mesonbuild/compilers/fortran.py

```diff
@@ -494,12 +494,24 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
                                  is_cross, info, exe_wrapper, linker=linker,
                                  full_version=full_version)
         self.id = 'nagfor'
+        self.warn_args = {
+            '0': ['-w=all'],
+            '1': [],
```

Comment:
So I take it that it reports all warnings by default?

Reply:
Yes that's correct.
(It also always reports a banner, which is not so great in these kinds of build systems. The forthcoming 7.1 release will add a -quiet option, hence the get_always_args change.)

Reply:
0 means no warnings, and 3 means "even useless warnings", i think

Reply:
Part of the nagfor design is as a linter in a way; (most) warnings are on by default. To switch off you add "-w=all: suppresses all warning messages". There could be (later) scope for adding at level 3 "-info: Request output of information messages. The default is to suppress these messages" if desired.

Reply:
Ah, okay. This is the opposite of GCC/clang where -Wall turns on a lot of warnings. Could you add a comment to the code so someone will meaning doesn't break this down the road

Reply:
In gcc the lowercase -w is for disabling warnings. Syntax is a bit different but the capitalization is a point of commonality.